CppCoreGuidelines.md

```diff
@@ -13138,12 +13138,21 @@ C++ provides better type checking and more notational support.
 It provides better support for high-level programming and often generates faster code.
 
 ##### Example
-
-    char ch = 7;
-    void* pv = &ch;
-    int* pi = pv;   // not C++
-    *pi = 999;      // overwrite sizeof(int) bytes near &ch
-
+    struct dataX{
+        char temperature;
+        char windspeed;
+    }
+    struct dataY{
+        int id;
+        double weight;
+    }
+    int main(){
+        struct dataX x;
+        void* ptrX = &x;
+        struct dataY* = ptrX; // not C++
+        dataY.temperature=20;// overwrite sizeof(int) bytes near &temperature, 
+        //thus corrupting "windspeed" and other data or structure padding
```

Comment:
It seems that even in C implicit casting from void * to a struct * is illegal. 
Also i seem to forget to name the variable dataY y.
 I guess a simple example with an extra function makes it clear why not to use a void pointer. 

```
#include <iostream>

  struct dataX{
      char temperature;
      char windspeed;
  };

  struct dataY{
      float weight;
  };

  //no type checking
  void dataEditing(void* dataPtr)
  {
    struct dataY* y = (dataY*) dataPtr;
    y->weight = -20000.08;// overwrite sizeof(int) bytes near &temperature,
    //thus corrupting "windspeed" and other data or structure padding
  }

  int main(){
    struct dataX x;
    dataEditing(&x);// not C++
    std::cout<< std::hex << x.temperature << "   " << x.windspeed  ;

  }

```

or the old example wit the char and int should be swapped around to you cast an int to a char pointer. since this would avoid memory miss alignment. or just an int to void* to float (which still geve problems on exotic ISAs with a seperate float memory).